src/niraj/narcissisticNumbers.spec.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  isNarcissisticNumber,
+  narcissisticNumbers
+} from './narcissisticNumbers';
+
+test('0 is not a narcissistic number', () => {
+  expect(isNarcissisticNumber(0)).toBe(false);
+});
+
+test('11 is not a narcissistic number', () => {
+  expect(isNarcissisticNumber(11)).toBe(false);
+});
+
+test('1 is a narcissistic number', () => {
+  expect(isNarcissisticNumber(1)).toBe(true);
+});
+
+test('153 is a narcissistic number', () => {
+  expect(isNarcissisticNumber(153)).toBe(true);
+});
+
+describe('Given a negative starting number', () => {
+  test('it should return an empty array', () => {
+    expect(narcissisticNumbers(-10).length).toBe(0);
+  });
+});
+
+describe('Given a negative ending number', () => {
+  test('it should return an empty array', () => {
+    expect(narcissisticNumbers(0, -10).length).toBe(0);
+  });
+});
+
+describe('Given a negative range', () => {
+  test('it should return an empty array', () => {
+    expect(narcissisticNumbers(-20, -10).length).toBe(0);
+  });
+});
+
+describe('When no range given', () => {
+  test('it should return an empty array', () => {
```

Comment:
Isn't the purpose of `describe` is to group the tests and have some common scope and it's own `beforeEach` and `afterEach` than using it for each tests. Having `describe` for each test won't give any advantage right?

Reply:
https://jestjs.io/docs/en/setup-teardown#scoping

Reply:
I could have written the tests without 'describe', but adding 'describe' helped my test suits look good

Reply:
> Since all the tests are related to testing a single functionality, can write a single describe block to enclose all the tests.

Perhaps, Can you please give recommendation on how to arrange my tests in logical grouping?

Reply:
how about something like this?
```
describe('Narcissistic number', () => {
   test('0 is not a narcissistic number', () => {... });
   test('it should return an empty array for a negative starting number input', () => {... });
   ...
});
```

Reply:
describe can be used to group test cases of similar kinds. In this case @nirajunify has created a detailed test cases, each one representing different types or edge cases. May be we can group those things into a nested describe such as negative numbers in one group
`describe('Given a negative number', () => {
  describe('Given a negative starting number', () => {
    test('it should return an empty array', () => {
      expect(narcissisticNumbers(-10).length).toBe(0);
    });
  });

  describe('Given a negative ending number', () => {
    test('it should return an empty array', () => {
      expect(narcissisticNumbers(0, -10).length).toBe(0);
    });
  });
});`

Or when we are testing for different range of positive numbers like this

`describe('When starting number is 10 and ending number is 10000', () => {
  const narcissisticNumbersList = [153, 370, 371, 407, 1634, 8208, 9474];
  test('it should return an array of numbers [153, 370, 371, 407, 1634, 8208, 9474]', () => {
    expect(narcissisticNumbers(10, 10000).length).toEqual(
      narcissisticNumbersList.length
    );
    expect(narcissisticNumbers(10, 10000)).toEqual(narcissisticNumbersList);
  });

  test('it should return an array of numbers [153, 370, 371, 407]', () => {
    expect(narcissisticNumbers(10, 500).length).toEqual(
      narcissisticNumbersList.splice(0,4).length
    );
    expect(narcissisticNumbers(10, 500)).toEqual(narcissisticNumbersList.splice(0,4));
  });
});`

Reply:
Thank you @krajasekhar-unify , @ramjayunify  for your valuable feedback. I will update the tests

Reply:
I think in each describe group there must be at least one pass case and one fail case .

Reply:
Thanks @ramjayunify for the `kentcdodds` article. I am convinced to not group the tests with describe at all. Do we have anything to inspire from the article and define in our guidelines?

Reply:
Thanks. I will update the specs